php/commands/package.php

```diff
@@ -183,6 +183,14 @@ public function install( $args, $assoc_args ) {
 		$json_manipulator->addMainKey( 'name', 'wp-cli/wp-cli' );
 		$json_manipulator->addLink( 'require', $package_name, $version );
 		$json_manipulator->addConfigSetting( 'secure-http', true );
+
+		$composer_backup_decoded = json_decode( $composer_backup, true );
```

Comment:
Nope.  It doesn't even decode it internally, it uses all regex for manipulating it.